user_resources.py

```
@@ -82,20 +83,62 @@ def __init__(self, username, password, tenant_id=None, tenant_name=None,
                 'Either tenant_id or tenant_name or trust_id must be provided')
-
+        region = self.clients.region
+        self.clients.override_endpoint(
+            'identity', region, 'admin', settings.KEYSTONE_ENDPOINT)
         self.user_id = self.clients.get_session().get_user_id()
         session = self.clients.get_session()
         self.user_name = session.auth.get_access(session)
         self.nova = NovaResources(self.clients)
         self.cinder = CinderResources(self.clients)
         self.glance = GlanceResources(self.clients)
-        self.neutron = NeutronResources(self.clients)
+        try:
+            self.neutron = NeutronResources(self.clients)
+        except Exception:
+            self.neutron = None
         self.blueprints = BluePrintResources(self.clients)
-        self.swift = SwiftResources(self.clients)
+        try:
+            self.swift = SwiftResources(self.clients)
+        except Exception:
+            self.swift = None
         # Images in use is a set used to avoid deleting formerly glance images
         # in use by other tenants
         self.imagesinuse = set()
 
+        # Regions the user has access
+        self.regions_available = set()
+        self.regions_available.update(self.clients.get_regions('compute'))
+
+    def change_region(self, region):
+        """
+        change the region. All the clients need to be updated, but the
+        session does not.
+        :param region: the name of the region
+        :return: nothing.
+        """
+        self.clients.set_region(region)
+        self.clients.override_endpoint(
+            'identity', region, 'admin', settings.KEYSTONE_ENDPOINT)
+        self.nova.on_region_changed()
+        self.glance.on_region_changed()
+        try:
+            if self.swift:
+                self.swift.on_region_changed()
+            else:
+                self.swift = SwiftResources(self.clients)
+        except Exception:
+            self.swift = None
```

do you not need to raise an exception?